### PR TITLE
Added function to create initial files

### LINKED ISSUE #20: Create project creation function
Create a function that will take required parameters such as project name, type of project (Node,python etc.) and will create a folder as well as run appropriate initialization function, such as npm init or cargo init etc. This folder should be created in a centralized space, particular to our app.
Make sure that you test this function, and create mocha tests for making sure that this runs, with proper harness and mocking of required data.

### PR DESCRIPTION
**Pull request by raj2729**
## Description

- Created a function to store respective projects inside a Projects Directory at location : 
```C:\Users\[USER_NAME]\AppData\Roaming\ergo``` 

- Created test for the same
 
## Changes

- Added file ```createProject.js``` at location ```app/electron/```
- Added file ```createProject.js``` at location ```test/```
- Added sample creation of project at ```main.js```
- Modified ```initial.js``` to also store information whether ```django``` is installed or not which may be used later
- Changed the version of shelljs to its previous version due to some errors in the latest version
- Increased the time limit for mocha tests to 20000ms as the creation of the folder require more time than the default 2000ms

## Issue

> Closes #20 

## Impacted Area

> The ```config.json``` file containing information during the initialization of the electron app is stored at :
```C:\Users\[USER_NAME]\AppData\Roaming\ergo```

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

- Call the function with the required parameters

image

### CODE REVIEW

File: app/electron/createProject.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      log.info("Projects Folder successfully created");
+    }
+  });
+
+  if (project.type === "node") {
+    // Traversing to inside the Projects Folder
+    cmd = `mkdir ${project.name} && cd ${project.name} && npm init -y`;
+  } else if (project.type === "django") {
+    cmd = `django-admin startproject ${project.name} && cd ${project.name} && echo > requirements.txt`;
+  }
```
Comment (YJDoc2) [inline app/electron/createProject.js:29]: Take this as param rather than determining in the function

File: app/electron/createProject.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      `${location[0]}: && cd ${location} && cd Projects && ${cmd}`,
+      { silent: true },
+      (code, stdout, stderr) => {
+        if (code !== 0) {
+          log.error(stderr);
+          reject(stderr);
+        } else if (code === 0) {
+          log.info(`Successfully created Project : ${project.name}`);
+          resolve({ success: true });
+        }
```
Comment (YJDoc2) [inline app/electron/createProject.js:42]: Will this work on Linux as well?

File: test/createProject.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    try {
+      let result = await createProject(project2 , location)
+      expect(result).to.have.property("success");
+      expect(result.success).to.equal(true);
+    } catch (error) {
+      expect(error).to.be.a('string');
+      expect(error).to.include('exists');
+    }
+  });
+});
```
Comment (YJDoc2) [inline test/createProject.js:36]: Add tests for Linux as well

**Review by YJDoc2 — APPROVED** [review]
Some changes are required

File: app/electron/createProject.js (hunk 1 above)
Comment (raj2729) [reply app/electron/createProject.js:29]: Could you please elaborate on what and how exactly are we planning to take data from the frontend?

File: app/electron/createProject.js (hunk 2 above)
Comment (raj2729) [reply app/electron/createProject.js:42]: No, it only works for windows

**Review by namandangi — APPROVED** [review]

**Pull request merged**